scripts/workspace-conversion/README.md

```diff
@@ -0,0 +1,61 @@
+## Converting Workspace Dataset Attributes to Catalog Entry Data
+
+# Description
+This script takes the workspace dataset attributes and attempts to map them to similar fields in the data catalog service as catalog entries.
+
+# Dataset Attributes
+An example workspace with attributes can be found [here](https://github.com/DataBiosphere/terra-ui/blob/dev/src/data/workspace-attributes.js#2), under the variable `displayLibraryAttributes`.
+
+The request made to retrieve the workspace details is an XHR request to rawls to retrieve the workspace information. The basic structure of the return of the call looks like:
+
+```
+{
+  owners: [],
+  workspaces: {
+    attributes: {
+      description: string,
+      `library:cellType: string,
+      `library: ...
+    }
+  }
+}
+```
+
+What is shown to the user in that table is the contents in the attributes that terra-ui has determined is important to display. These values are hardcoded at 
+`{terra-ui}/src/data/workspace-cleanAttributes.js.displayLibraryAttributes`
+
+# Value mappings
+To run the conversion from workspace to dataset, here is a current mapping scheme:
+
+
+| i | Attribute | Rawls Workspace Path | Data Catalog Entry Path |
+| - | --------- | -------------------- | -------------------------- |
+| 1 | Cohort Phenotype/Indication (Disease Ontology)  | `library:diseaseOntologyLabel` | `samples.disease.0`<br>kathy: lists predefined terms, but forces them to stick with a specific vocabulary. The other phenotype field is the diseases |
+| 2 | No. of Subjects | `library:numSubjects` | `counts.donors`<br>Kathy: suggests leaving out this field because it is volatile and can easily become inaccurate |
+| 3 | Data Category | `library:dataCategory`<br>ex: "Simple Nucleotide Variation, Copy Number Variation, Expression Quantification, DNA-Methylation, Clinical phenotypes, Biosample metadata" | `dct:dataCategory`*<br>This field does not currently exist |
+| 4 | Experimental Strategy | `library:datatype`<br>ex: "Whole Exome, Genotyping Array, RNA-Seq, miRNA-Seq, Methylation Array, Protein Expression Array" | Kathy: Map this to data modality, talk to kathy about getting the mapping of terms |
+| 5 | Data Use Limitation | `library:dataUseRestriction` | `TerraDCAT_ap:hasDataUsePermission.0`<br>Not a perfect mapping, we will need to normalize "General Research Use" to "TerraCore:NoRestriction", preferably using the mapping found [here](https://github.com/DataBiosphere/terra-ui/blob/dev/src/pages/library/dataBrowser-utils.js#23) under `datasetReleasePolicies` |
+| 6 | Cohort Phenotype/Indication | `library:indication`<br>ex: "Mesothelioma" | kathy: might not be useful, could consider pre-pending with source (ie: "TCGA" or "Anvil_phenotype") |
```

Comment:
Looking at the examples, I think this corresponds to disease.  We don't usually store that on a dataset because many datasets will have many diseases.  Again, we could add it as a field but long term, would we calculate it or do we need to provide a caveat that this data was submitted with the original dataset and may not include all diseases represented in the dataset.  You can add it as a field but it won't be part of TIM.

Reply:
We will add it as a value in the diseases array, and fix it later after we can examine some more of the generated data